### DISCUSSION TITLE
NX + Nix = 🩵

### DISCUSSION THREAD

**Discussion by gnuion** [nrwl/nx#23117] [General]
I recently decided to try to contribute to the NX codebase, but the following error got yelled:
```sh
/bin/sh: rustc: command not found
```

So why not use Nix to solve system dependencies? Sure, Nix does not work on Windows, but who develops on Windows anyways?

**Comment by MilanKovacic** [comment]
> but who develops on Windows anyways?

A lot of people. Specifically, around 50% of people according to the latest Stack Overflow survey: https://survey.stackoverflow.co/2023/#section-most-popular-technologies-operating-system

**Reply by gnuion** [reply]
@MilanKovacic yeah on WSL. My experience with working with containers in Windows is terrible.

**Reply by MilanKovacic** [reply]
No. As you can see from the survey, ~15% of people use WSL.
I use Mac, but Windows is still the largest operating system by market share, both in personal, and professional use.

**Reply by jaysoo** [reply]
We do need to support Windows. The Nx core team members all have Windows machines to test on, and if it's not possible to contribute to Nx using Windows, it'll make debugging/fixing Windows issues that much harder.

**Comment by jaysoo** [comment]
The `CONTRIBUTING.md` should provide all the information needed. If something is missing you are welcome to create a PR to add it. 

The rust part in particular is called out specifically here: https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#building-the-project

**Reply by gnuion** [reply]
Right. There are similarities between NX and Nix though. Both are all about caching and reproducibility. I like them both. To install rust on a machine that uses Nix, I can just use a flake with direnv, or install it with home-manager. Best thing is, all builds are cached and will be pulled, so I don't need to build packages from source. Similar how NX works.

**Reply by jaysoo** [reply]
I'll make a note of it, and if others are interested please upvote this discussion. There are other priority items that the team is working on, but we're open to new ideas.
